File: PI2CabelloLinaresAntonio/src/ejercicios/ejercicio3RecursivoNoFinal.java
package ejercicios;

public class ejercicio3RecursivoNoFinal {

	public static Long potenciaRecursivoNoFinal1(Long a, Integer n) {
		return potenciaRecursivoNoFinal(a, n);
	}

	private static Long potenciaRecursivoNoFinal(Long a, Integer n) {

		long res = 1;

		if (n > 0) {

			if (n % 2 == 1) {

				res = potenciaRecursivoNoFinal(a * a, n / 2) * a;// (a^n/2)^2*a
			}

			else if (n % 2 == 0) {
				res = potenciaRecursivoNoFinal(a * a, n / 2);// (a^n/2)^2
			}
		}
		// sino res = 1 (n==0)

		return res;
	}

}

File: PI2CabelloLinaresAntonio(C)/ejercicios/ejercicio3RecursivoNoFinal.c
/*
 * ejercicio3RecursivoNoFinal.c
 *
 *  Created on: 21 nov. 2020
 *      Author: anton
 */


#include "ejercicio3RecursivoNoFinal.h"

long potenciaRecursivoNoFinal1(long a, int n){

	return potenciaRecursivoNoFinal(a, n);
}

long potenciaRecursivoNoFinal(long a, int n){

	long res = 1;

	if(n>0){

		if(n%2==1){
			return potenciaRecursivoNoFinal(a*a, n/2)*a;// (a^n/2)^2*a
		}

		else if(n%2==0){
			return potenciaRecursivoNoFinal(a * a, n / 2);// (a^n/2)^2
		}
	}

	//sino res = 1 (n==0)

	return res;
}
